pootle/core/models/snapshot.py

```diff
@@ -90,8 +92,10 @@ def save(self, ctx_data):
         except OSError:
             pass
 
-        with open(self.filepath, 'w') as outfile:
-            outfile.write(ctx_data)
+        # use newline='' not to convert \n to OS-specific newlines;
+        # this is to be able to generate snapshots on any platform
+        with io.open(self.filepath, 'w', newline='') as outfile:
+            outfile.write(unicode(ctx_data))
```

Comment:
`ctx_data` is the data already serialized to JSON, which are bytes. Opening the file in binary mode (`wb`) allows to write bytes to the file, hence avoiding the extra conversion to the `unicode` type.